### PR TITLE
All code changes up till now

### PR DESCRIPTION
**Pull request by CSdread**
This PR includes:
* refactoring of major parts of the code into packages
* Building of a output writer factory system to make it easy to create new output writers
* adding unit tests and updating unit tests for all changes
* building an HTTP Writer that sends generic HTTP POST calls
* A Notification service writer built on top of the HTTPWriter
* Adding a different kind of message filter on top of what they already have, that allows us to filter messages based on their key (using the `-k` argument when creating the rule)
* Changing references to point to `pantheon-systems` go-audit instead of `slackhq`
* CI/DI Pipeline using circle ci with release to github releases on our fork

### CODE REVIEW

File: audit.go [hunk 1]
```diff
@@ -11,8 +11,8 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/slackhq/go-audit/pkg/output"
-	"github.com/slackhq/go-audit/pkg/slog"
+	"github.com/pantheon-systems/go-audit/pkg/output"
```
Comment (joemiller) [inline audit.go]: It's my understanding that re-writing paths on a fork is bad practice. It makes it much more difficult to integrate with upstream.

My understanding about how to handle this for local dev is to go-get the original and then add your fork, eg:
```
cd $GOPATH/src/slackhq/go-audit
git remote add fork github.com/pantheon-systems/go-audit
```

I have used that to contribute to a few projects and it works well enough. It also is the recommended way for go, AFAIK.

What I have not done tho is brought a fork into a full CI/CD pipeline. But I imagine the approach that might work there is similar -- have circle checkout to the upstream path, or `mv` it into it, then build it.

Comment (CSdread) [reply audit.go]: all of those packages are in our fork only though...

Comment (joemiller) [reply audit.go]: @CSdread I see. Thanks. We should not modify the vendor.json's rootpath then, if we intend to try to get anything in this repo upstreamed.

**Comment by joemiller** [comment]
What was the rationale for moving things into pkg/* dir in that slackhq PR?

**Comment by joemiller** [comment]
Can you provide some examples of how the new filtering with the -k flag works?

**Comment by CSdread** [comment]
the reason for the pkg/* refactor is that I wanted to write our code cleanly, so I put the output factories and output functions into the package, then i had to do the same with some of the output systems dependencies cause main cannot rely on output and output rely on main. So things started moving into the packages dir. The output plugins before were hard-coded in the main `audit.go` file and had no pluggable structure. The alternate to the pkg/* of the code base was basically to have no fork and to hard code our changes to the output right in `audit.go`

**Comment by joemiller** [comment]
lgtm 👍

File: audit.go (hunk 1 above)
Comment (CSdread) [reply audit.go]: @joemiller i will research this and get it done

**Pull request merged**